src/query/models/tags_test.go

```diff
@@ -22,78 +22,136 @@ package models
 
 import (
 	"bytes"
+	"fmt"
 	"hash/fnv"
 	"reflect"
 	"testing"
 	"unsafe"
 
+	"github.com/m3db/m3/src/query/util/writer"
 	xtest "github.com/m3db/m3/src/x/test"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func createTags(withName bool) Tags {
-	tags := NewTags(3, nil).AddTags([]Tag{
+func testLongTagIDOutOfOrder(t *testing.T, scheme IDSchemeType) Tags {
+	opts := NewTagOptions().SetIDSchemeType(scheme)
+	tags := NewTags(3, opts).AddTags([]Tag{
 		{Name: []byte("t1"), Value: []byte("v1")},
+		{Name: []byte("t3"), Value: []byte("v3")},
 		{Name: []byte("t2"), Value: []byte("v2")},
+		{Name: []byte("t4"), Value: []byte("v4")},
 	})
 
-	if withName {
-		tags = tags.SetName([]byte("v0"))
-	}
-
 	return tags
 }
 
-func TestTagID(t *testing.T) {
-	tags := createTags(false)
-	assert.Equal(t, "t1=v1,t2=v2,", tags.ID())
-	assert.Equal(t, tags.IDLen(), len(tags.ID()))
+func TestLongTagNewIDOutOfOrderLegacy(t *testing.T) {
+	tags := testLongTagIDOutOfOrder(t, TypeLegacy)
+	actual := tags.ID()
+	assert.Equal(t, tags.idLen(), len(actual))
+	assert.Equal(t, []byte("t1=v1,t2=v2,t3=v3,t4=v4,"), actual)
 }
 
-func TestTagIDMarshalTo(t *testing.T) {
-	var (
-		tags = createTags(false)
-		b    = tags.IDMarshalTo([]byte{})
-	)
-	assert.Equal(t, []byte("t1=v1,t2=v2,"), b)
-	assert.Equal(t, tags.IDLen(), len(b))
+func TestLongTagNewIDOutOfOrderQuoted(t *testing.T) {
+	tags := testLongTagIDOutOfOrder(t, TypeQuoted)
+	actual := tags.ID()
+	assert.Equal(t, []byte(`t1="v1",t2="v2",t3="v3",t4="v4"`), actual)
```

Comment:
nit: Just as discussed, to exact match the Prometheus ID construction, adding the `{` and `}` should get us 100% matching.

Reply:
I think the only reason we wouldn't want to do that, is if M3QL ends up using the same ID format. Which it probably should TBH.

Reply:
Should be fixed 👍